testing/trino-tests/src/test/java/io/trino/tests/ci/TestCiWorkflow.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.tests.ci;
+
+import org.junit.jupiter.api.Test;
+import org.yaml.snakeyaml.Yaml;
+
+import java.io.StringReader;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import static com.google.common.collect.ImmutableList.toImmutableList;
+import static com.google.common.collect.ImmutableSet.toImmutableSet;
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class TestCiWorkflow
+{
+    private static final Path CI_YML = Paths.get("../../.github/workflows/ci.yml");
```

Comment:
Hard-coded or relative paths in tests are brittle. They depend on what’s the working directory when the test is executed.

Also, tests within a module should not rely on anything outside the module (except for maven dependencies)

Reply:
Agreed with both points. This is really poorly written. I wish i knew how to do this better.

How would you like to see it, given the tested thing is not a maven module?

Reply:
> Does the brittleness of the test justify the benefit?

Absolutely. Otherwise i wouldn't create it.

Reply:
Can you describe the benefit it provides over the guarantees that GH actions makes? What are you seeing that I'm not?

Reply:
GH actions don't make sufficient promises. Thus it is possible to add new job to ci workflow without updating build-success's definition and thus making build-success result incorrect.

With a test it is not possible, unless the test isn't run (impact analysis). But  we run all tests on .github/workflow changes and we always will be, so impact analysis won't skip this test. Even if we somehow skip the test, we run all tests on master, so we will notice the problem rather sooner than later (compared to noticing rather later than sooner if we don't have a test).

Test brittleness. Agreed. If we assume the working directory can be _anything_, even finding repo's top level (the folder when `.git/` is anchored) is impossible. However, most tests are run from command line (`mvn test -pl :module-name`) or from IDE (typically Intellij). The test currently passes in both scenarios.